test/types.ts

```diff
@@ -1,9 +1,13 @@
+interface ExtendedArray<T> extends Array<T> {}
```

Comment:
I suggest reverting changes in this file because the following rule has to be true: `ComplexNestedRequired = DeepRequired<ComplexNestedPartial>` and `ComplexNestedRequired = DeepWritable<ComplexNestedReadonly>`

readonly and extended arrays are an exception to these rules so has to be handled separately

Reply:
Done!